pkg/devfile/parser/data/v2/volumes_test.go

```diff
@@ -175,248 +155,157 @@ func TestDevfile200_AddVolume(t *testing.T) {
 				},
 			}
 
-			err := d.AddVolume(tt.args.volumeComponent, tt.args.path)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("AddVolume() error = %v, wantErr %v", err, tt.wantErr)
-			}
-
-			if err != nil && tt.wantErr {
-				return
-			}
-
-			if !reflect.DeepEqual(d.Components, tt.wantComponents) {
-				t.Errorf("wanted: %v, got: %v, difference at %v", tt.wantComponents, d.Components, pretty.Compare(tt.wantComponents, d.Components))
+			err := d.AddVolumeMount(tt.args.componentName, tt.args.name, tt.args.path)
+			if tt.wantErr && err == nil {
```

Comment:
why split this logic `(err != nil) != tt.wantErr` into two if-else blocks? 
the existing one is clear enough to me

Same for other same changes.